src/auth/s2s/serviceAuth.ts

```diff
@@ -0,0 +1,19 @@
+import { totp } from 'node-otp'
+import { http } from '../../http/http'
+import { S2SConfig } from './s2sConfig.interface'
+
+// TODO: To be replaced with a proper logging library
+const logger = console
+
+export async function postS2SLease(s2sConfig: S2SConfig): Promise<string> {
```

Comment:
Why is this is a separate exported function instead of part of the s2s class that's created?

Reply:
This function has now been moved into the main `s2s.class.ts` file.